org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/debug/firefox/Tester.java

```diff
@@ -0,0 +1,47 @@
+package org.eclipse.wildwebdeveloper.debug.firefox;
+
+import org.eclipse.core.expressions.PropertyTester;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.core.resources.IResource;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.IAdaptable;
+
+public class Tester extends PropertyTester {
+	private static final String PROPERTY_NAME = "isHTMLProject"; //$NON-NLS-1$
+	@Override
+	public boolean test(Object receiver, String property, Object[] args, Object expectedValue) {
+		if (property.equals(PROPERTY_NAME)) {
+			IResource resource = toResource(receiver);
+			if (resource == null) {
+				return false;
+			}
+			return isHTMLProject(resource.getProject());
+		}
+		return false;
+	}
+
+	private IResource toResource(Object o) {
+		if (o instanceof IResource) {
+			return (IResource) o;
+		} else if (o instanceof IAdaptable) {
+			return ((IAdaptable) o).getAdapter(IResource.class);
+		} else {
+			return null;
+		}
+	}
+
+	public static boolean isHTMLProject(IProject p) {
+		if (p == null || !p.isAccessible()) {
+			return false;
+		}
+		try {
+			for (IResource projItem : p.members()) {
+				if (projItem.getName().equals("index.html")) { //$NON-NLS-1$ 
```

Comment:
The launch shortcut doesn't (yet) contain logic about detecting the `index.html`. Is this logic included in the language servers themselves? Ie, if provided a folder, do they automatically start the `index.html` ? That would actually be great and may automatically get things working better in Wild Web Developer.

Reply:
I don't think the debug adapter (I assume this is what you mean by language server) has any logic to automatically start the `index.html`. In my testing, trying to open a folder in Firefox presents the user with a directory explorer.

The logic in `Tester.java` is only used to determine if it makes sense for a file or project to be debuged in Firefox. 

The actual launch shortcut will most likely need logic to find the `index.html` and set that as the `file` parameter in `FirefoxRunDABDebugDelegate`.

Reply:
> The actual launch shortcut will most likely need logic to find the index.html and set that as the file parameter in FirefoxRunDABDebugDelegate.

Exactly, and I propose we add this logic later to focus on getting the shortcut shown only where it works at the moment (`.html` files).

Reply:
Agreed. Working on it now :)